dace/codegen/sdfgcc.py

```diff
@@ -4,24 +4,39 @@
 import dace
 import os
 import sys
+import argparse
+import shutil
 
 
 def main():
-    if len(sys.argv) != 2:
-        print('USAGE: sdfgcc <PATH TO SDFG FILE>')
-        exit(1)
+    # Command line options parser
+    parser = argparse.ArgumentParser(description='Simple SDFG command-line compiler.')
+
+    # Required argument for SDGF file path
+    parser.add_argument('SDFGfilepath', help='<PATH TO SDFG FILE>', type=str)
+
+    # Optional argument for output location
+    parser.add_argument('-o','--out', type=str, help='If provided, saves lib and header file to path. Directories in path need to exist beforehand.')
 
-    filename = sys.argv[1]
-    if not os.path.isfile(filename):
-        print('SDFG file', filename, 'not found')
-        exit(2)
+    args = parser.parse_args()
+
+    filepath = args.SDFGfilepath
+    if not os.path.isfile(filepath):
+        print('SDFG file', filepath, 'not found')
+        exit(1)
 
     # Load SDFG
-    sdfg = dace.SDFG.from_file(filename)
+    sdfg = dace.SDFG.from_file(filepath)
 
     # Compile SDFG
-    sdfg.compile()
+    sdfg.compile(args.out)
+
+    # Copying header file to optional path
+    if args.out:
```

Comment:
What if the argument is a filename? If it's a folder we use it as is, if it's a file we pass it along to the compiler (it will rename to that file) and should use its base path to store the .h file there.